File: mobile/src/main/java/io/ashdavies/cinnamon/home/HomePresenter.java
package io.ashdavies.cinnamon.home;

import io.ashdavies.cinnamon.presenter.AbstractViewPresenter;
import javax.inject.Inject;

class HomePresenter extends AbstractViewPresenter<HomeView> {

  @Inject
  HomePresenter(HomeView view) {
    super(view);
  }
}

File: mobile/src/main/kotlin/io/ashdavies/cinnamon/home/HomePresenter.kt
package io.ashdavies.cinnamon.home

import io.ashdavies.cinnamon.presenter.AbstractViewPresenter
import io.ashdavies.cinnamon.trades.Client
import javax.inject.Inject

internal class HomePresenter @Inject constructor(client: Client, view: HomeView) : AbstractViewPresenter<HomeView>(view) {

  init {
    client.fetch().subscribe(view::onNext, view::onError)
  }
}
